processor/k8sattributesprocessor/internal/kube/automatic.go

```diff
@@ -0,0 +1,26 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package kube // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sattributesprocessor/internal/kube"
+
+import (
+	"regexp"
+)
+
+type AutomaticRules struct {
+	Enabled            bool     `mapstructure:"enabled"`
+	AnnotationPrefixes []string `mapstructure:"annotation_prefixes"`
+}
+
+const (
+	DefaultAnnotationPrefix = "resource.opentelemetry.io/"
+)
+
+func AutomaticAnnotationRule(prefix string) FieldExtractionRule {
+	return FieldExtractionRule{
+		Name:                 "$1",
+		KeyRegex:             regexp.MustCompile(`^` + regexp.QuoteMeta(prefix) + `(.+)$`),
+		HasKeyRegexReference: true,
+		From:                 MetadataFromPod,
+	}
```

Comment:
This can be just a variable. No need for a function and passing the const

```
var defaultOtelAnnotationExtractionRule = FieldExtractionRule{
		Name:                 "$1",
		KeyRegex:             regexp.MustCompile(`^` + regexp.QuoteMeta(defaultOtelAnnotationPrefix) + `(.+)$`),
		HasKeyRegexReference: true,
		From:                 MetadataFromPod,
	}
```

Can probably be moved to processor/k8sattributesprocessor/options.go or even right inside `withOtelAnnotations`.